src/Toml/PrefixTree.hs

```diff
@@ -96,10 +96,9 @@ type PrefixMap a = HashMap Piece (PrefixTree a)
 -- | Data structure to represent table tree for @toml@.
 data PrefixTree a
     = Leaf !Key !a
```

Comment:
Let's also comment fields of  `Leaf` constructor while we're on it. Actually, we can make `tomland` quite an excellent library if we add Haddock comments as much as possible just like you did in `KeysDiff` data type.